components/eam/src/physics/crm/crm_history.F90

```diff
@@ -443,126 +447,129 @@ subroutine crm_history_out(state, ptend, crm_state, crm_rad, crm_output, crm_ecp
    ! CRM_QRS + CRM_QRL output in radiation_tend will show a time lag.
 
    ! GCM level rad heating tendencies
-   call outfld('MMF_QRL   ',qrl/cpair, pcols, lchnk )
-   call outfld('MMF_QRS   ',qrs/cpair, pcols, lchnk )
+   call outfld('MMF_QRL   ',qrl/cpair, ncol, lchnk )
```

Comment:
I think we’ve looked at this probably half a dozen times, and I still get confused about outfld dimensions, but I don’t think you want to do this here, unless you explicitly make qrl and qrs have size ncol as passed to outfld. I think the size of the array or pointer you pass to outfld needs to match the value of the dimension you pass, or weird behavior can result. I.e., I think you could do:
```
call outfld(‘MMF_QRL’, qrl(1:ncol,:)/cpair, ncol, lchnk)
```
or
```
call outfld(‘MMF_QRL’, qrl/cpair, pcols, lchnk)
```
but not what you have changed this to.

Reply:
Also side note, why are we outputting these here? This same field is also output in radiation.F90, with the name QRS, QRL. This appears to be the same field, so we now have three versions of the heating rate: CRM_QRL (the crm-resolved heating), QRL (the gridmean heating rate), and MMF_QRL (also the gridmean heating rate). It looks like this has been in here, but why?

Reply:
I'm not sure why we are outputting these here, I'm ok to get rid of them. I didn't realize they were a duplicate. 

But to your first point this is actually a very important change because the data is ONLY ncol values because we are working with data where each set of "ncol" values has been stapled together across all chunks on a given task. So I think it's only correct to use ncol rather than pcols here. I can add the explicit dimensions though, that's not a bad idea.